File: laoo_api/Models/MasterConstCodes.cs
namespace LaooApi.Models;

/// Central constants stored in TDSTMasterConst.
public static class MasterConstCodes
{
    public const string CustomerCodeGeneration = "001";
    public const string ItemCodeGeneration = "002";

    public const string ConstRunCus = "ConstRunCus";
    public const string ConstRunItem = "ConstRunItem";

    public const string Manual = "0";
    public const string RunByYear = "1";
    public const string RunAuto = "2";
    public const string RunByItemType = "1";

    public static readonly IReadOnlyDictionary<string, string> GroupVariables =
        new Dictionary<string, string>(StringComparer.Ordinal)
        {
            [CustomerCodeGeneration] = ConstRunCus,
            [ItemCodeGeneration] = ConstRunItem,
        };
}

File: lib/core/master/master_const_codes.dart
/// Central constants stored in TDSTMasterConst.
abstract final class MasterConstCodes {
  static const customerCodeGeneration = '001';
  static const itemCodeGeneration = '002';

  static const constRunCus = 'ConstRunCus';
  static const constRunItem = 'ConstRunItem';

  static const manual = '0';
  static const runByYear = '1';
  static const runAuto = '2';
  static const runByItemType = '1';

  static const groupVariables = <String, String>{
    customerCodeGeneration: constRunCus,
    itemCodeGeneration: constRunItem,
  };

  static const customerCodeOptions = <String, String>{
    manual: 'ใส่เองอิสระ',
    runByYear: 'Run ภายใต้ปี',
    runAuto: 'Run Auto',
  };

  static const itemCodeOptions = <String, String>{
    manual: 'ใส่เองอิสระ',
    runByItemType: 'Run ภายใต้ประเภทสินค้า',
    runAuto: 'Run Auto',
  };
}
